### PR TITLE
Add filter to disable gzip file creation

### LINKED ISSUE #197: add option to disable GZ compression for HDD caching
Related to request from WP support forums: https://wordpress.org/support/topic/disable-gz/

The HDD caching module always generates a static HTML file along with a gzipped version:
https://github.com/pluginkollektiv/cachify/blob/dcac6b7babdef4454782faf6f9c076699cb11944/inc/cachify_hdd.class.php#L165-L166

We have a corresponding block in the Apache HTTP server example using _mod_mime_ to server the pre-compressed content.

With most webservers supporting compression natively which can be superior to compression by PHP and more important is not handled during initial site visit, many sites don't need the explicit _*.gz_ files.

We should at least introduce an option to disable generation of pre-compressed files to reduce the overhead here.

### PR DESCRIPTION
**Pull request by angcl**
@florianbrinkmann and I worked on this PR that adds a filter that allows to disable gzip file creation.

Fixes #197

### CODE REVIEW

File: inc/class-cachify-hdd.php [hunk 1]
```diff
@@ -170,6 +170,12 @@ private static function _create_files( $data ) {
 		}
 		/* Write to file */
 		self::_create_file( self::_file_html( $file_path ), $data );
+
+		/**
+		 * Filter that allows to enable/disable GZIP file creation
+		 *
+		 * @param bool $create_gzip_files Whether to create GZIP files. Default is `true`
```
Comment (pfefferle) [inline inc/class-cachify-hdd.php]: shouldn't it be `@return`?

**Review by pfefferle — APPROVED** [review]
LGTM!

**Comment by stklcode** [comment]
Filter looks good.

We should conditionally remove the GZip instructions from the .htaccess instructions in the settings tab, too, if generation is disabled, even though this should be obvious … but we all know copy&pasting is one thing, understanding another.

Did we discusse a checkbox in the settings that may address more users?

**Comment by florianbrinkmann** [comment]
I guess that is a feature that is only relevant for a small part of our users, so that a filter should be enough, but that is only a guess :)

**Comment by angcl** [comment]
gzip instructions now won't be added to .htaccess instructions if gzip file creation is disabled. :)

**Review by stklcode — APPROVED** [review]

**Pull request merged**